src/backend/distributed/commands/foreign_constraint.c

```diff
@@ -134,10 +134,6 @@ ErrorIfUnsupportedForeignConstraint(Relation relation, char distributionMethod,
 	int referencedColumnCount = 0;
 	bool isNull = false;
 	int attrIdx = 0;
```

Comment:
Seems that many of the remaining variables could also have gone into the loop, though they don't seem problematic.